frontend/device/android/device.lua

```diff
@@ -67,7 +67,9 @@ local Device = Generic:new{
     isAndroid = yes,
     model = android.prop.product,
     hasKeys = yes,
-    hasDPad = no,
+    hasDPad = function() return android.isTv() end,
+    hasFewKeys = function() return android.isTv() end,
```

Comment:
Aha! I knew it would pay off someday. :-)

(I'd think TV remotes had more buttons though?)

Reply:
The documentation says: https://developer.android.com/training/tv/start/controllers

So we have 6 buttons. (4 dpad, back and select). I use `hasFewKeys` as it opens the top menu in the reader with the right dpad button. Some things are still rough (ie: we don't need to close the menu with left dpad button, back is ok; there's no way to open top menu in the FM...)

Reply:
I see. On Kindle etc. we have a menu key for the top menu and an ok/select/whatever key for the bottom menu. Just a menu key lacking is a lot less bad, but kind of annoying that it's not sufficiently full-featured.

> there's no way to open top menu in the FM

Perhaps something could be done with left and right, e.g., by reversing it from the normal FM button layout. Left for context menu, right for top menu.

Reply:
My suggestion would be:

## Reader
![dpad-reader](https://user-images.githubusercontent.com/975883/86772057-9a31e080-c053-11ea-91d5-5cac4341907a.jpg)

## File manager
![dpad-fm](https://user-images.githubusercontent.com/975883/86772074-a322b200-c053-11ea-83a1-6d45826859a3.jpg)

## Menus
![dpad-menu](https://user-images.githubusercontent.com/975883/86772094-ab7aed00-c053-11ea-9223-ebdbbb300189.jpg)

Reply:
The reason I suggested right for top menu in the file manager is so it'd be consistent with the reader, or was that one for the bottom menu? We should try to keep it internally consistent with the reader too, within reason. Overall the above proposal looks good! :+1:

Reply:
Right now on the reader the center d-pad button opens the bottom menu. On the FM the right dpad button opens file menu.

Reply:
In that case I added → for the top menu (on hasFewKeys).

Reply:
That works on the reader, not on the filemanager, where it opens selected file/directory properties.